internal/dao/secret.go

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of K9s
+
+package dao
+
+import (
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Secret represents a secret K8s resource.
+type Secret struct {
+	Resource
+	decode bool
+}
+
+// Describe describes a secret that can be encoded or decoded.
+func (s *Secret) Describe(path string) (string, error) {
+	encodedDescription, err := s.Resource.Describe(path)
+
+	if err != nil {
+		return "", err
+	}
+
+	if !s.decode {
+		return encodedDescription, nil
+	}
+
+	return s.Decode(encodedDescription, path)
+}
+
+// SetDecode sets the decode flag.
+func (s *Secret) SetDecode(flag bool) {
+	s.decode = flag
+}
+
+// Decode removes the ecnode part from the secret's description and appends the
+// secret's decoded data.
+func (s *Secret) Decode(encodedDescription, path string) (string, error) {
+	o, err := s.GetFactory().Get(s.GVR(), path, true, labels.Everything())
+
+	if err != nil {
+		return "", err
+	}
+
+	dataEndIndex := strings.Index(encodedDescription, "====") + 4
+
+	if dataEndIndex == -1 {
```

Comment:
This likely will never fire since we add 4 - Better check the actual index call

Reply:
indeed. good catch ^^'
fixed